src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,746 @@
+
+  GEOSX_THROW_IF( cycleNumber > sourceValue.size( 0 ), "Too many steps compared to array size", std::runtime_error );
+  forAll< serialPolicy >( m_sourceConstants_x.size( 0 ), [=] ( localIndex const isrc )
+  {
+    if( sourceIsLocal[isrc] == 1 )
+    {
+      for( localIndex inode = 0; inode < m_sourceConstants_x.size( 1 ); ++inode )
+      {
+        rhs_x[sourceNodeIds[isrc][inode]] = sourceConstants_x[isrc][inode] * sourceValue[cycleNumber][isrc];
+        rhs_y[sourceNodeIds[isrc][inode]] = sourceConstants_y[isrc][inode] * sourceValue[cycleNumber][isrc];
+        rhs_z[sourceNodeIds[isrc][inode]] = sourceConstants_z[isrc][inode] * sourceValue[cycleNumber][isrc];
+      }
+    }
+  } );
+}
+
+void ElasticWaveEquationSEM::computeSeismoTrace( real64 const time_n,
+                                                 real64 const dt,
+                                                 real64 const timeSeismo,
+                                                 localIndex iSeismo,
+                                                 arrayView1d< real64 const > const var_np1,
+                                                 arrayView1d< real64 const > const var_n,
+                                                 arrayView2d< real64 > varAtReceivers )
+{
+  real64 const time_np1 = time_n+dt;
+  arrayView2d< localIndex const > const receiverNodeIds = m_receiverNodeIds.toViewConst();
+  arrayView2d< real64 const > const receiverConstants   = m_receiverConstants.toViewConst();
+  arrayView1d< localIndex const > const receiverIsLocal = m_receiverIsLocal.toViewConst();
+
+  real64 const a1 = (dt < epsilonLoc) ? 1.0 : (time_np1 - timeSeismo)/dt;
+  real64 const a2 = 1.0 - a1;
+
+  if( m_nsamplesSeismoTrace > 0 )
+  {
+    forAll< EXEC_POLICY >( receiverConstants.size( 0 ), [=] GEOSX_HOST_DEVICE ( localIndex const ircv )
+    {
+      if( receiverIsLocal[ircv] == 1 )
+      {
+        varAtReceivers[iSeismo][ircv] = 0.0;
+        real64 vtmp_np1 = 0.0;
+        real64 vtmp_n = 0.0;
+        for( localIndex inode = 0; inode < receiverConstants.size( 1 ); ++inode )
+        {
+          vtmp_np1 += var_np1[receiverNodeIds[ircv][inode]] * receiverConstants[ircv][inode];
+          vtmp_n += var_n[receiverNodeIds[ircv][inode]] * receiverConstants[ircv][inode];
+        }
+        // linear interpolation between the pressure value at time_n and time_(n+1)
+        varAtReceivers[iSeismo][ircv] = a1*vtmp_n + a2*vtmp_np1;
+      }
+    } );
+  }
+
+  // TODO DEBUG: the following output is only temporary until our wave propagation kernels are finalized.
+  // Output will then only be done via the previous code.
+  if( iSeismo == m_nsamplesSeismoTrace - 1 )
+  {
+    forAll< serialPolicy >( receiverConstants.size( 0 ), [=] ( localIndex const ircv )
+    {
+      if( this->m_outputSeismoTrace == 1 )
+      {
+        if( receiverIsLocal[ircv] == 1 )
+        {
+          // Note: this "manual" output to file is temporary
```

Comment:
You can remove these two comments, we already have a `TODO` above saying the same thing